### PR TITLE
 Class music Album

### PR DESCRIPTION
**Pull request by cvalencia1991**
# in this milestone, i did the following activite 

- Create Music Album Class

### CODE REVIEW

File: MusicAlbum/musicalbum.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+include '../items'
+
+class MusicAlbum
+  attr_reader :archived
+
+  def initialize(on_spotify: true)
+    @on_spotify = on_spotify
+    super(archived)
```
Comment (KanzaTahreem) [inline MusicAlbum/musicalbum.rb]: Archieved is not defined in the method. If you want to pass it as a parameter to initialize function than it should be added to parameter list. If you intend to use @archived attribute it is not nil when an instance is created for a class untill some value is assigned.
Also super(archived) means you are calling initiallize function of parent class. But you haven't inherited MusicAlbum from Item class as shown in the UML diagram. First inherit from item and then call super function with the parms that initialize function of Item class has. Either the Item class has initialize function something like this:
```
  def initialize(id, date, archived: false)
```
So, super should be 
```
super(id, date, archived: archived)
```

File: MusicAlbum/musicalbum.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  end
+
+  private
+
+  def can_be_archived?
+    true if @archived == true && @on_spotify == true
```
Comment (KanzaTahreem) [inline MusicAlbum/musicalbum.rb]: As described in documentation call the parent class can_be_archived here. So this line should be something like this:
```
super && @on_spotify = true
```

File: MusicAlbum/musicalbum.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+include '../items'
+
+class MusicAlbum
+  attr_reader :archived
+
+  def initialize(on_spotify: true)
```
Comment (KanzaTahreem) [inline MusicAlbum/musicalbum.rb]: As mentioned earlier, it think we should add id here reference to the constructor we are using, something like mentioned below
```
def initialize(date, archived: archived, on_spotify, id: nil)
  super (id, date, archived: archived)
  @on_spotify = on_spotify
end
```

Also, set getter and setter for on_spotify not for archived

File: MusicAlbum/musicalbum.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+include '../items'
+include '../Genre/genre'
+
+class MusicAlbum < Genre
```
Comment (KanzaTahreem) [inline MusicAlbum/musicalbum.rb]: Inherit Music Album from Item class not from Genre

**Review by KanzaTahreem — CHANGES_REQUESTED** [review]
# Changes Required ♻
Hi @cvalencia1991 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! 
**_However, I did notice a few areas where some additional changes would further enhance the quality of the code._** 
But you are almost there! 

_I am confident that addressing these concerns/making these changes will bring the code to an even higher level of excellence and it will be better than before 💪._

almost-there

## Highlights
There are some excellent points that you have made perfectly. Keep up the great work! 👏

- Well done so far
- Good commit messages

However, there is always room for improvement.

## Required Changes

> _Please keep in mind that your long hours of work is appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄_

_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------

File: MusicAlbum/musicalbum.rb (hunk 2 above)
Comment (cvalencia1991) [reply MusicAlbum/musicalbum.rb]: ok you right im going to change it

File: MusicAlbum/musicalbum.rb (hunk 1 above)
Comment (cvalencia1991) [reply MusicAlbum/musicalbum.rb]: yes that why i never understad it, thank you im going to change it

File: MusicAlbum/musicalbum.rb (hunk 3 above)
Comment (cvalencia1991) [reply MusicAlbum/musicalbum.rb]: yes im going to change it

File: MusicAlbum/musicalbum.rb (hunk 4 above)
Comment (cvalencia1991) [reply MusicAlbum/musicalbum.rb]: yes im going to change it

**Review by KanzaTahreem — APPROVED** [review]
# Approved 🥳

Hi @cvalencia1991 

Great work nailing the project 💪

**_Your project is COMPLETE! There is nothing else to say other than... it's time to merge it! 🚀 Congratulations!_**



## Highlights
There are some excellent points that you have made perfectly. Keep up the great work! 👏
- Great work using the right Git Flow  👍
- Commits are descriptive, good work.
- Linters are working properly, great job! ✅

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------

**Comment by cvalencia1991** [comment]
thank you very much 😄

**Pull request merged**